doc/rvv-intrinsic-spec.adoc

```diff
@@ -431,52 +431,195 @@ The mask types encode the ratio that is derived from `EEW`/`EMUL`. The mask type
+| `vfloat64m1_t` | `vfloat64m1x2_t` | `vfloat64m1x3_t` | `vfloat64m1x4_t` | `vfloat64m1x5_t` | `vfloat64m1x6_t` | `vfloat64m1x7_t` | `vfloat64m1x8_t`
+|===
+
+.Tuple types (EMUL=2)
+[cols="2,2,2,2,2,2,2,2"]
+[%autowidth]
+|===
+| Non-tuple Types (NFILED=1) | NFIELD=2   | NFIELD=3    | NFIELD=4     | NFIELD=5     | NFIELD=6   | NFIELD=7 | NFIELD=8
+| `vint8m2_t` | `vint8m2x2_t` | `vint8m2x3_t` | `vint8m2x4_t` | N/A | N/A | N/A | N/A
+| `vuint8m2_t` | `vuint8m2x2_t` | `vuint8m2x3_t` | `vuint8m2x4_t` | N/A | N/A | N/A | N/A
+| `vint16m2_t` | `vint16m2x2_t` | `vint16m2x3_t` | `vint16m2x4_t` | N/A | N/A | N/A | N/A
+| `vuint16m2_t` | `vuint16m2x2_t` | `vuint16m2x3_t` | `vuint16m2x4_t` | N/A | N/A | N/A | N/A
+| `vint32m2_t` | `vint32m2x2_t` | `vint32m2x3_t` | `vint32m2x4_t` | N/A | N/A | N/A | N/A
+| `vuint32m2_t` | `vuint32m2x2_t` | `vuint32m2x3_t` | `vuint32m2x4_t` | N/A | N/A | N/A | N/A
+| `vint64m2_t` | `vint64m2x2_t` | `vint64m2x3_t` | `vint64m2x4_t` | N/A | N/A | N/A | N/A
+| `vuint64m2_t` | `vuint64m2x2_t` | `vuint64m2x3_t` | `vuint64m2x4_t` | N/A | N/A | N/A | N/A
+| `vfloat16m2_t` | `vfloat16m2x2_t` | `vfloat16m2x3_t` | `vfloat16m2x4_t` | N/A | N/A | N/A | N/A
+| `vfloat32m2_t` | `vfloat32m2x2_t` | `vfloat32m2x3_t` | `vfloat32m2x4_t` | N/A | N/A | N/A | N/A
+| `vfloat64m2_t` | `vfloat64m2x2_t` | `vfloat64m2x3_t` | `vfloat64m2x4_t` | N/A | N/A | N/A | N/A
+|===
+
+.Tuple types (EMUL=4)
+[cols="2,2,2,2,2,2,2,2"]
+[%autowidth]
+|===
+| Non-tuple Types (NFILED=1) | NFIELD=2   | NFIELD=3    | NFIELD=4     | NFIELD=5     | NFIELD=6   | NFIELD=7 | NFIELD=8
+| `vint8m4_t` | `vint8m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vuint8m4_t` | `vuint8m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vint16m4_t` | `vint16m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vuint16m4_t` | `vuint16m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vint32m4_t` | `vint32m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vuint32m4_t` | `vuint32m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vint64m4_t` | `vint64m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vuint64m4_t` | `vuint64m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vfloat16m4_t` | `vfloat16m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vfloat32m4_t` | `vfloat32m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+| `vfloat64m4_t` | `vfloat64m4x2_t` | N/A | N/A | N/A | N/A | N/A | N/A
+|===
+
+
+[[pseudo-intrinsics]]
 == Pseudo intrinsics
 
 The intrinsics provide extra utility functions to help users manipulate across the RVV intrinsic types. These functions are called "pseudo intrinsics". These pseudo intrinsics do not represent any real instructions.
 
-[[pseudo-vsetvl-vsetvlmax]]
-=== `vsetvl`/`vsetvlmax`
+[[pseudo-vsetvl]]
+=== `vsetvl`
+
+The `vsetvl` intrinsics return `min(avl, VLMAX)` given the provided element width and LMUL in the suffix. Please refer to how VLMAX is defined in the v-spec ^5^.
```

Comment:
`min(avl, VLMAX)` doesn't match `vset{i}vl{i}` in the V spec, where for `VLMAX < avx < 2*VLMAX` it can be anything in `[ceil(VLMAX/2), VLMAX]`

Reply:
Added description to clarify the semantics. Please check again, thank you.